src/google-classroom-extractor/google_classroom_extractor/api/usage.py

```diff
@@ -0,0 +1,123 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from datetime import datetime, timedelta
+import logging
+import os
+from typing import List, Dict, Optional
+from dateutil.parser import parse as date_parse
+import pandas as pd
+from sqlalchemy.exc import OperationalError
+from .api_caller import call_api
+
+
+def request_usage(resource, date: datetime) -> List[Dict[str, str]]:
+    return call_api(
+        resource.userUsageReport().get,
+        {  # type: ignore
+            "userKey": "all",
+            "date": date,
+            "parameters": "classroom:timestamp_last_interaction,classroom:num_posts_created,accounts:timestamp_last_login",
+        },
+        "usageReports",
+    )
+
+
+def last_sync_date(sync_db) -> Optional[datetime]:
+    with sync_db.connect() as con:
+        try:
+            usage_df = pd.read_sql("SELECT asOfDate FROM Usage", con)
+            if usage_df["asOfDate"].count() == 0:
+                return None
+            return date_parse(usage_df["asOfDate"].max())
+        except OperationalError:
+            # no Usage table yet
```

Comment:
Could add a logging.error

Reply:
It's actually acceptable behavior when there is no table in db.  Can add a logging.debug().